### DISCUSSION TITLE
Is it possible for darling to support #[rename_all(serialize="snake_case", deserialize="camelCase")]

### DISCUSSION THREAD

**Discussion by pymongo** [poem-web/poem#51] [Ideas]
serde support both `#[rename_all="snake_case"]` and `#[rename_all(serialize="snake_case", deserialize="camelCase")]`

here is my try: https://github.com/pymongo/poem/commit/edf5dc596ac3b0fb6cc97ab482ea625461a95862

```rust
#[Derive(Object)]
#[rename_all(serialize="snake_case", deserialize="camelCase")]
struct A;
```

But I think darling is impossible to support:

```rust
#[derive(Debug, Copy, Clone, FromMeta)]
pub(crate) enum RenameAllRule {
    RenameRule(RenameRule),
    Serde {
        serialize: Option<RenameRule>,
        deserialize: Option<RenameRule>
    }
}
```

**Comment by sunli829** [comment]
Yes, you need to implement `FromMeta` yourself, the `FromMeta` macro does not support this.
